app/javascript/packages/document-capture/components/acuant-capture.jsx

```diff
@@ -290,6 +316,7 @@ function AcuantCapture(
 
       isForceUploading.current = false;
     } else {
+      isProxyingClickEvent.current = true;
```

Comment:
I'm not quite following why we have to set this here, can you explain?

Reply:
> I'm not quite following why we have to set this here, can you explain?

The problem is that when a user clicks on the "Take Picture" or "Upload" buttons, we handle this by emulating a click on the file input, which is required so that the browser native behavior for file input happens normally. But since the file input is the placeholder, and the placeholder's click behavior is to log a `source: 'placeholder'` click event, without this logic, we'd have two events being logged for "Take Picture" or "Upload" button clicks: one for `source: 'button'` or `source: 'upload'`, but then a second unexpected `source: 'placeholder'`.

Basically, the code here is saying "disregard the next click that's about to happen", so that the second event isn't logged.

I noticed this pretty late, and in retrospect it could be avoided by logging the events inline, but (a) it's non-ideal as far as code duplication and (b) we might still have to deal with it in this function anyways, since the "Take Picture" button will eventually cause both top-level condition paths of this `startCaptureOrTriggerUpload` to be run.

Reply:
Or as represented by failing test case when this line is removed:

```
     AssertionError: expected stub to have been called exactly thrice, but it was called 4 times
    stub({ label: "IdV: test image clicked", payload: { source: "placeholder" } }) at /Users/andrewmduthie/Documents/Code/identity-idp/app/javascript/packages/document-capture/components/acuant-capture.jsx:276:9
    stub({ label: "IdV: test image clicked", payload: { source: "button" } }) at /Users/andrewmduthie/Documents/Code/identity-idp/app/javascript/packages/document-capture/components/acuant-capture.jsx:276:9
    stub({ label: "IdV: test image clicked", payload: { source: "placeholder" } }) at /Users/andrewmduthie/Documents/Code/identity-idp/app/javascript/packages/document-capture/components/acuant-capture.jsx:276:9
    stub({ label: "IdV: test image clicked", payload: { source: "upload" } }) at /Users/andrewmduthie/Documents/Code/identity-idp/app/javascript/packages/document-capture/components/acuant-capture.jsx:276:9
```

Reply:
Ok thanks, I think I get it now.

What about another wrapper function to wrap these simulated clicks, so then it can set the variable as needed?

```js
simulateClick(inputRef.current)
```

or a wrapper variety:

```js
skipClickLogging(() =>
  inputRef.current?.click()
);
```

Reply:
> What about another wrapper function to wrap these simulated clicks, so then it can set the variable as needed?
> 
> ```js
> simulateClick(inputRef.current)
> ```

Yeah, I like that idea 👍  Also a bit more self-documenting, which is nice.

Reply:
Updated in 882d4a61f.